frontend/javascripts/admin/tasktype/task_type_create_view.js

```diff
@@ -110,8 +110,13 @@ class TaskTypeCreateView extends React.PureComponent<Props, State> {
       formValues.recommendedConfiguration = defaultValues.recommendedConfiguration;
     }
     formValues.recommendedConfiguration = jsonStringify(formValues.recommendedConfiguration);
-    formValues.settings.allowedMagnificationsmin = formValues.settings.allowedMagnifications.min;
-    formValues.settings.allowedMagnificationsmax = formValues.settings.allowedMagnifications.max;
+    formValues.settings.resolutionRestrictionsForm = {
+      shouldRestrict:
+        formValues.settings.resolutionRestrictions.min != null ||
+        formValues.settings.resolutionRestrictions.max != null,
+      min: formValues.settings.resolutionRestrictions.min || 1,
+      max: formValues.settings.resolutionRestrictions.max || 512,
+    };
```

Comment:
I like that you separated the form vs not-form aspects explicitly here. However, I'm wondering whether we could do it without having both values simultaneously in the `formValues`?

Something like this:
```
function getFormValuesForTaskType() {
  const taskType = _.merge({}, defaultValues, taskType);
  formValues.recommendedConfiguration = jsonStringify(formValues.recommendedConfiguration);

  // explain here what's going on
  const resolutionRestrictions = taskType.settings.resolutionRestrictions;
  delete taskType.settings.resolutionRestrictions;
  taskType.settings.resolutionRestrictionsForm = ...;
}

const formValues = getFormValuesForTaskType(taskType)
```